cmd/snap/cmd_assert_build.go

```diff
@@ -0,0 +1,140 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	_ "golang.org/x/crypto/sha3" // expected for digests
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/i18n"
+)
+
+type cmdAssertBuild struct {
+	Positional struct {
+		Filename string `positional-arg-name:"<filename>" description:"filename of the snap you want to assert a build for"`
+	} `positional-args:"yes" required:"yes"`
+
+	DeveloperID string `long:"developer-id" description:"identifier of the signer" required:"yes"`
+	SnapID      string `long:"snap-id" description:"identifier of the snap package associated with the build" required:"yes"`
+	KeyName     string `long:"key-name" description:"name of the GnuPG key to use (otherwise 'default' is assumed)"`
+	Grade       string `long:"grade" description:"grade states the build quality of the snap: <stable|devel> (defaults to 'stable')"`
+}
+
+var shortAssertBuildHelp = i18n.G("Process a snap file and assert its build")
+var longAssertBuildHelp = i18n.G(`
+Mainly used to generate and sign snap-build assertions at the moment.
+`)
+
+func init() {
+	cmd := addCommand("assert-build",
```

Comment:
Both sound like valid perspectives. The outcome is indeed an assertion, but in terms of the meaning of that to the developer, most people would think about it as signing that particular build to prove that that's where it was built. Assertion is the mechanism.. signing that file as coming from that developer is the intention. 

On that basis, I'd prefer sign-build as well, as it's more meaningful to the person calling it.